webapp/tests/model/test_recursive_data.py

```diff
@@ -0,0 +1,172 @@
+        input_data = {'can_fly': True, 'pet_name': 'Yoshi'}
+        try:
+            self.TwoPreviousMockRecursiveDataModel.parse_obj(input_data)
+        except ValidationError:
+            self.fail('TwoPreviousMockRecursiveDataModel.parse_obj should not raise a validation error')
+
+    def test_if_two_previous_fields_and_first_missing_and_second_prev_missing_then_raise_validation_error(self):
+        input_data = {'pet_name': 'Yoshi'}
+        self.assertRaises(ValidationError, self.TwoPreviousMockRecursiveDataModel.parse_obj, input_data)
+
+    def test_if_two_previous_fields_and_second_missing_and_first_correct_then_raise_no_validation_error(self):
+        input_data = {'can_jump': True}
+        try:
+            self.TwoPreviousMockRecursiveDataModel.parse_obj(input_data)
+        except ValidationError:
+            self.fail('TwoPreviousMockRecursiveDataModel.parse_obj should not raise a validation error')
+
+    def test_if_two_previous_fields_and_second_prev_field_missing_and_first_correct_then_raise_no_validation_error(self):
+        input_data = {'can_jump': True, 'pet_name': 'Yoshi'}
+        try:
+            self.TwoPreviousMockRecursiveDataModel.parse_obj(input_data)
+        except ValidationError:
+            self.fail('TwoPreviousMockRecursiveDataModel.parse_obj should not raise a validation error')
+
+    def test_if_two_previous_fields_and_both_missing_then_raise_validation_error(self):
+        input_data = {}
+        self.assertRaises(ValidationError, self.TwoPreviousMockRecursiveDataModel.parse_obj, input_data)
+
+    def test_if_two_previous_fields_and_first_raises_value_error_and_second_not_then_raise_no_error(self):
+        input_data = {'can_jump': True, 'can_fly': True, 'pet_name': 'Yoshi'}
+        BaseMario.should_raise_error = True
+        try:
+            self.TwoPreviousMockRecursiveDataModel.parse_obj(input_data)
+        except ValidationError:
+            self.fail('TwoPreviousMockRecursiveDataModel.parse_obj should not raise a validation error')
+
+    def test_if_two_previous_fields_and_second_raises_value_error_and_first_not_then_raise_no_error(self):
+        input_data = {'can_jump': True, 'can_fly': True, 'pet_name': 'Yoshi'}
+        CapeMario.should_raise_error = True
+        try:
+            self.TwoPreviousMockRecursiveDataModel.parse_obj(input_data)
+        except ValidationError:
+            self.fail('TwoPreviousMockRecursiveDataModel.parse_obj should not raise a validation error')
+
+    def test_if_two_previous_fields_and_both_raise_value_error_then_raise_validation_error(self):
+        input_data = {'can_jump': True, 'can_fly': True, 'pet_name': 'Yoshi'}
+        CapeMario.should_raise_error = True
+        BaseMario.should_raise_error = True
+        self.assertRaises(ValidationError, self.TwoPreviousMockRecursiveDataModel.parse_obj, input_data)
+
+    def tearDown(self) -> None:
+        BaseMario.should_raise_error = False
+        CapeMario.should_raise_error = False
```

Comment:
The problem is that we have to set it before the first run and we want to set it back to false after every run. We could get rid of the tearDown call. However, it might be in an incorrect state afterwards. If you have a better solution to tell the class that the exception should be raised I would love to improve that.